staging/src/slime.io/slime/modules/limiter/controllers/envoy_filters.go

```diff
@@ -180,22 +188,60 @@ func descriptorsToGlobalRateLimit(descriptors []*microservicev1alpha2.SmartLimit
 	return generateGlobalRateLimitDescriptor(globalDescriptors, loc)
 }
 
-// TODO get labels base on serviceentry
 func generateServiceSelector(r *SmartLimiterReconciler, spec *LimiterSpec) map[string]string {
-	var labels map[string]string
-	// top level
 	if len(spec.labels) > 0 {
-		labels = spec.labels
+		// top level
+		return spec.labels
+	} else if spec.host != "" {
+		// second level
+		nn := types.NamespacedName{Namespace: spec.loc.Namespace, Name: spec.host}
+		svc := getIstioService(r, nn)
```

Comment:
when the svc arrives later than smartlimiter. how to handle this case?

Reply:
ticker mechanism will ensure module get the svc